### PR TITLE
Add Dump offload support using http dynamic response

### PR DESCRIPTION
**Pull request by raviteja-b**
This commit
1.Enables dynamic response support
2.Defines new bmcweb rule to support dump streaming. 3.Implemented routes for bmc and system dump offload.

Tested By:
1. BMC dump offload
2. System dump offload

### CODE REVIEW

**Comment by sunharis** [comment]
This needs clang formatting @raviteja-b

**Comment by raviteja-b** [comment]
> This needs clang formatting @raviteja-b

Ran Clang and updated PR

**Comment by raviteja-b** [comment]
I see CI failures power_supply.hpp file
```
core/lib/power_supply.hpp:425:23: error: the 'empty' method should be used to check for emptiness instead of 'size' [readability-container-size-empty,-warnings-as-errors]
            if (ec || object.size() == 0)
                      ^~~~~~~~~~~~~~~~~~
                      object.empty()
/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_vector.h:1083:7: note: method 'vector'::empty() defined here
      empty() const _GLIBCXX_NOEXCEPT
```
     this is not related to this commit

**Comment by gtmills** [comment]
@raviteja-b What upstream commits are these?

**Comment by gtmills** [comment]
jenkins run tests please

**Comment by raviteja-b** [comment]
> @raviteja-b What upstream commits are these?

Upstream gerrit link
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/38051

**Review by sunharis — APPROVED** [review]
Upstream code did not get approvals from the maintainers. This is the downstream only code now. Looks good.

File: http/http_stream.hpp
Comment (gtmills) [inline http/http_stream.hpp:87]: #include <string>

File: http/http_stream.hpp
Comment (gtmills) [inline http/http_stream.hpp:104]: #include <boost/system/error_code.hpp>

File: include/dump_offload.hpp
Comment (gtmills) [inline include/dump_offload.hpp:24]:
```
#include <string>
```

File: include/dump_offload.hpp
Comment (gtmills) [inline include/dump_offload.hpp:339]: This is really a CRITICAL log? CRITICAL is supposed to be bmcweb crashed and such

File: include/dump_offload.hpp
Comment (gtmills) [inline include/dump_offload.hpp:402]: Really CRITICAL?

File: include/dump_offload.hpp
Comment (gtmills) [inline include/dump_offload.hpp:495]: Maybe atleast lower these to ERROR but really should be Debug or Info..

File: http/http_stream.hpp
Comment (raviteja-b) [reply http/http_stream.hpp:87]: do we need to include this header file?
its getting compiled right?

File: http/http_stream.hpp
Comment (raviteja-b) [reply http/http_stream.hpp:104]: why do we need this?

File: include/dump_offload.hpp
Comment (raviteja-b) [reply include/dump_offload.hpp:339]: yes correct, but we need to this info log in journal traces.
so we have decided to use CRITICAL + INFO: for debug traces

File: include/dump_offload.hpp
Comment (raviteja-b) [reply include/dump_offload.hpp:402]: yeah these are really needed to debug @gtmills 
we have added these traces as per HMC -BMC issues we have seen

File: include/dump_offload.hpp
Comment (raviteja-b) [reply include/dump_offload.hpp:495]: I don't see debug or INFO log in journal, so we have decided to use CRITICAL + "INFO"

File: http/http_stream.hpp
Comment (gtmills) [reply http/http_stream.hpp:104]: https://www.incredibuild.com/blog/include-what-you-use-how-to-best-utilize-this-tool-and-avoid-common-issues

File: http/http_stream.hpp
Comment (gtmills) [reply http/http_stream.hpp:87]: Yes.

**Comment by sunharis** [comment]
Until we have a good debug facility / error log facility at bmcweb, we will have to use this way to get our debug data. This code is in 1020 and 1030 as well. 
These traces are hit only when there is a dump offload

File: include/dump_offload.hpp
Comment (raviteja-b) [reply include/dump_offload.hpp:24]: is there any change in string usage in coding guidelines?

Comment (gtmills) [reply include/dump_offload.hpp:24]: You should include what you use

File: http/http_stream.hpp
Comment (raviteja-b) [reply http/http_stream.hpp:90]: done

File: http/http_stream.hpp
Comment (raviteja-b) [reply http/http_stream.hpp:107]: done

File: include/dump_offload.hpp
Comment (raviteja-b) [reply include/dump_offload.hpp:25]: done

File: include/dump_offload.hpp
Comment (gtmills) [inline include/dump_offload.hpp:264]: The new code all uses    sdbusplus::asio::getProperty
https://github.com/ibm-openbmc/bmcweb/blob/1050/redfish-core/lib/systems.hpp#L208

**Review by gtmills — APPROVED** [review]

**Pull request merged**
